QQE-1584.md

```diff
@@ -0,0 +1,82 @@
+# QQE-1584 Quarkus LangChain4J testing
+
+JIRA link: https://issues.redhat.com/browse/QQE-1584
+
+This is a high-level strategy for testing of langchain4j
+
+## Scope of the testing
+@SystemMessage + @UserMessage annotations
+RAG functionality
+Integration with MCP API
+
+### Optional, but desirable:
+@Tool + @RegisterAiService (only supported for OpenAI, so cannot be tested locally)
+Observability for tool requests (to avoid a situation, where LLM overwhelms the database with too many requests)
+
+## Existing coverage upstream:
+### Workshop
+https://quarkus.io/quarkus-workshop-langchain4j/step-11/#running-the-llm-inference-locally  and  https://github.com/quarkusio/quarkus-workshop-langchain4j/tree/main/step-11
+(they are not run automatically, as far as I can tell)
+The tests use jlama, which requires Java 21+
+Attempt to run the test locally with jlama fails on my laptop(32.0 GiB RAM, 13th Gen Intel® Core™ i7-1365U × 12). Will try to run it on server, but I am not very optimistic about that.
+Successfully ran it on the server. The downside is there is no rest api, and we have to connect through websocket (which is not optimal). Also, it hangs regularly, even on xlarge.
```

Comment:
Since it seems that our future testing hinges on using this repository, what is the strategy here to actually make it useful for our usage and to run with platforms we certify RHBQ with?